mono/metadata/object-internals.h

```diff
@@ -2160,6 +2160,8 @@ mono_string_empty_handle (MonoDomain *domain);
 gpointer
 mono_object_get_data (MonoObject *o);
 
+#define mono_get_data_unsafe(handle) ((guint8*)MONO_HANDLE_RAW (handle) + MONO_ABI_SIZEOF (MonoObject))
```

Comment:
Not clear what type this ought to return.
1. void*
2. char*
3. uint8*?
4. typeof(MONO_HANDLE_RAW(handle))?
5. uintptr


Current guint8* just bleeds from implementation.

Reply:
nit: mention `valuetype` somewhere in the name? Don't we have some function that does this already?

For C, probably `void*` made the most sense; in C++, `char*` is probably what we'll want.

Reply:
The other is right above `mono_object_get_data`. Nothing about `valuetype`. Or `unbox`ing.
I find `get_data` pretty non specific, but it is there.
I could say `mono_handle_get_data`, but `handle` is supposed to be somewhat superfluous. So is `object`. Hey, what else is mono dealing with, but objects and handles all over the place?

I'm flexible here.

Really in C++ -- overload, it just remains called the same old `mono_object_get_data` -- *and less churn* and no loss of clarity -- you don't really care if it is object or handle, it just works and does essentially the same thing.